source.go

```diff
@@ -226,11 +232,13 @@ var (
 type LLBGetter func(sOpts SourceOpts, opts ...llb.ConstraintsOpt) (llb.State, error)
 
 type ForwarderFunc func(llb.State, *SourceBuild) (llb.State, error)
+type GitCredHelperGetter func() (llb.State, error)
 
 type SourceOpts struct {
-	Resolver   llb.ImageMetaResolver
-	Forward    ForwarderFunc
-	GetContext func(string, ...llb.LocalOption) (*llb.State, error)
+	Resolver             llb.ImageMetaResolver
+	Forward              ForwarderFunc
+	GetContext           func(string, ...llb.LocalOption) (*llb.State, error)
+	GitCredentialHelpers map[string]GitCredHelperGetter
```

Comment:
Shouldn't there only ever be one helper here and it doesn't seem like it should be tied to any particular need (e.g. gomod).
This is how we make our buildkit secrets available to git, shouldn't need much more than that.

Reply:
Changed this to a `func() (llb.RunOption, error)`. The error is thrown if the frontend state can't be acquired. The runopt mounts the frontend at `/usr/local/bin/frontend` and can be used anywhere the credential helper is needed.